src/containers/UsernameTranslation.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Link } from "react-router-dom";
+
+/*
+At the moment works ONLY with apiPath='seacat_auth'
+- userId (required, array) - user ids to be translated
+- apiPath - default (and only working so far) 'seacat_auth'
+- cleanupTime - default set to 24 hours
+
+Usage:
+
+import { Credentials } from 'asab-webui';
+...
+<Credentials app={props.app} id={['user1-id', 'user2-id', 'user3-id']} apiPath='seacat_auth' cleanupTime={1000 * 60 * 60 * 24 * 14}/>
+
+*/
+
+export function Credentials(props) {
+
+	const { t } = useTranslation();
+
+	const apiPath = props.apiPath ?? 'seacat_auth';
+	let API = props.app.axiosCreate(apiPath);
+
+	const userIdsArray = Array.isArray(props.userId) ? props.userId : [props.userId] ;
+
+	const cleanupTime = props.cleanupTime ?? 1000 * 60 * 60 * 24; // 24 hrs
+
+	const [usernames, setUsernames] = useState([]);
+
+	// asks the server for usernames, saves them to local storage and sets usernames to render
+	const retrieveUserNames = async () => {
+		try {
+			let response = await API.put(`usernames`, userIdsArray);
+			if (response.data.result !== "OK") {
+				throw new Error(t("ASABCredentials|Something went wrong, failed to fetch assigned credentials"));
+			}
+			const usernamesToLS = setUsernamesToLS(response.data.data, userIdsArray, cleanupTime);
+			setUsernames(usernamesToLS);
+		} catch (e) {
+			props.app.addAlert("warning", t("ASABCredentials|Something went wrong, failed to fetch assigned credentials"));
+		}
+	}
+
+	// compares array of IDs with data in localstorage
+	const matchUserIds = (credentials_id) => {
```

Comment:
@petrKavulok This is kinda confusing me, why is here used `credentials_id` as a props name,
```
const matchUserIds = (credentials_id) =>
```
when it is clearly an array of user IDs? You are passing an array:

https://github.com/TeskaLabs/asab-webui/blob/afe4f174a64b803e78e6cdd7fdcbcedc9ce5eaef/src/containers/UsernameTranslation.js#L69

`credentials_id` should be used for single ID

I suggest to unify the naming, I guess this was some misunderstanding in this conversation: https://github.com/TeskaLabs/asab-webui/pull/218#discussion_r773895258